component/prometheus/operator/common/component.go

```diff
@@ -106,18 +106,27 @@ func (c *Component) Update(args component.Arguments) error {
 	return nil
 }
 
+// NotifyClusterChange implements component.ClusterComponent.
+func (c *Component) NotifyClusterChange() {
+	c.mut.Lock()
+	defer c.mut.Unlock()
+
+	if !c.config.Clustering.Enabled {
+		return // no-op
+	}
+
+	// Schedule a reload so targets get redistributed.
+	select {
+	case c.onUpdate <- struct{}{}:
```

Comment:
This is a case where this change can probably simplify things a lot. Trying to differentiate a clustering update vs an argument update was a real pain. We can probably not send on this channel, and just call `c.manager.ClusteringUpdated()` directly, and simplify the logic in Run by a bit.